### PR TITLE
add multi-label support

### PR DESCRIPTION
**Pull request by louis-huang**
Hi I added support to allow label as a list. So we can support reading data with multiple labels. This can then solve https://github.com/ray-project/xgboost_ray/issues/286.
I verified new unit tests pass. Also test_matrix.py all pass with my local set up.
I verified locally by training a xgboost model with parquet data format, it works well. So far it should work well for parquet data format. Thank you!

### CODE REVIEW

**Comment by louis-huang** [comment]
I verified the change works with the blow code example:
```
from sklearn.datasets import make_multilabel_classification
import pandas as pd
import numpy as np
n_classes = 5
random_state = 0
X, y = make_multilabel_classification(n_samples=32, n_classes=5, n_labels=3, random_state=random_state)
features = [f"f{i}" for i in range(len(X[0]))]
labels = [f"label_{i}" for i in range(n_classes)]

X_df = pd.DataFrame(X, columns = features)
y_df = pd.DataFrame(y, columns = labels)
data = pd.concat([X_df, y_df], axis = 1)

data.to_parquet("~/Desktop/sample_data/data.parquet")

from xgboost_ray import RayDMatrix, RayParams, train, RayFileType
n_classes = 5
features = [f"f{i}" for i in range(20)]

labels = [f"label_{i}" for i in range(n_classes)]

training_data = "~/Desktop/sample_data"
train_set = RayDMatrix(training_data, labels, columns = features + labels, filetype=RayFileType.PARQUET)

evals_result = {}
bst = train(
    {
        "objective": "binary:logistic",
        "eval_metric": ["logloss", "error"],
        "random_state": random_state,
    },
    train_set,
    num_boost_round = 1,
    evals_result=evals_result,
    evals=[(train_set, "train")],
    verbose_eval=False,
    ray_params=RayParams(
        num_actors=1,  # Number of remote actors
        cpus_per_actor=1))

#bst.save_model("model.xgb")
#print("Final training error: {:.4f}".format(
#    evals_result["train"]["error"][-1]))

from xgboost_ray import predict
pred_ray = predict(bst, train_set, ray_params=RayParams(num_actors=1))
print(pred_ray)


import xgboost as xgb

clf = xgb.XGBClassifier(tree_method="hist", n_estimators = 1, random_state=0)
clf.fit(X, y)
expected = clf.predict_proba(X)

np.testing.assert_allclose(expected, pred_ray)

```

**Comment by heyitsmui** [comment]
@Yard1 can you help take a look when you get a chance? thanks!

File: xgboost_ray/data_sources/data_source.py [hunk 1]
```diff
@@ -118,12 +118,12 @@ def convert_to_series(data: Any) -> pd.Series:
     @classmethod
     def get_column(
         cls, data: pd.DataFrame, column: Any
-    ) -> Tuple[pd.Series, Optional[str]]:
+    ) -> Tuple[pd.Series, Optional[Union[str, List]]]:
```
Comment (heyitsmui) [inline xgboost_ray/data_sources/data_source.py:121]: should we open up a separate `get_columns(...)` instead of overloading this method?

**Review by Yard1 — APPROVED** [review]
LGTM, thanks! cc @krfricke

File: xgboost_ray/matrix.py [hunk 2]
```diff
@@ -406,7 +408,9 @@ def get_data_source(self) -> Type[DataSource]:
         ):  # noqa: E721:
             # Label is an object of a different type than the main data.
             # We have to make sure they are compatible
-            if not data_source.is_data_type(self.label):
+            # if it's a parquet data source and label is a list, then we consider it a multi-label data
+            if not data_source.is_data_type(self.label) \
+                and not (isinstance(self.label, List) and data_source.__name__ == "Parquet"):
```
Comment (Yard1) [inline xgboost_ray/matrix.py:413]:
```suggestion
            # if it's a parquet data source and label is a list, then we consider it a multi-label data
            if not data_source.is_data_type(self.label) \
                and not (isinstance(self.label, List) \
                and data_source.__name__ == "Parquet"):
```

File: xgboost_ray/tests/test_matrix.py [hunk 3]
```diff
@@ -276,6 +291,18 @@ def testFromMultiCSVString(self):
                 [data_file_1, data_file_2], "label", distributed=True
             )
 
+    def testFromParquetStringMultiLabel(self):
+        with tempfile.TemporaryDirectory() as dir:
+            data_file = os.path.join(dir, "data.parquet")
+
+            data_df = pd.DataFrame(self.x, columns=["a", "b", "c", "d"])
+            labels = [f"label_{l}" for l in range(4)]
```
Comment (Yard1) [inline xgboost_ray/tests/test_matrix.py:299]:
```suggestion
            labels = [f"label_{label}" for label in range(4)]
```

File: xgboost_ray/tests/test_matrix.py [hunk 4]
```diff
@@ -286,6 +313,28 @@ def testFromParquetString(self):
...
             self._testMatrixCreation(data_file, "label", distributed=False)
             self._testMatrixCreation(data_file, "label", distributed=True)
+    
+    def testFromMultiParquetStringMultiLabel(self):
+        with tempfile.TemporaryDirectory() as dir:
+            data_file_1 = os.path.join(dir, "data_1.parquet")
+            data_file_2 = os.path.join(dir, "data_2.parquet")
+
+            data_df = pd.DataFrame(self.x, columns=["a", "b", "c", "d"])
+            labels = [f"label_{l}" for l in range(4)]
```
Comment (Yard1) [inline xgboost_ray/tests/test_matrix.py:323]:
```suggestion
            labels = [f"label_{label}" for label in range(4)]
```

File: xgboost_ray/matrix.py [hunk 5]
```diff
@@ -406,7 +408,10 @@ def get_data_source(self) -> Type[DataSource]:
         ):  # noqa: E721:
             # Label is an object of a different type than the main data.
             # We have to make sure they are compatible
-            if not data_source.is_data_type(self.label):
+            # if it's a parquet data source and label is a list, then we consider it a multi-label data
```
Comment (Yard1) [inline xgboost_ray/matrix.py:411]:
```suggestion
            # if it's a parquet data source and label is a list,
            # then we consider it a multi-label data
```

**Comment by louis-huang** [comment]
Hi @Yard1 may I ask how to fix the lint test? Seems it still blocks the merge. Thank you!

**Comment by Yard1** [comment]
Can you run the ./format.sh script in the root of the repo?

**Comment by yc2984** [comment]
@louis-huang can you please run the above test please?
